src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -32,33 +44,136 @@ pub struct SealedSstableBuilder {
     pub bloom_filter_size: usize,
 }
 
+/// Consumer for the output of `SstableBuilder::finish`.
+#[async_trait::async_trait]
+pub trait SstableBuilderSealer<W>
```

Comment:
I think we don't need this trait. 

The current two implementations of this trait differs only in the way they finish the upload. Therefore, we can instead move the upload logic into the `finish` of the implementation of `SstableWriter`, and in the `seal_current` of `CapacitySplitSstableBuilder` we just simply put the output of writer to the `SealedSstableBuilder`.

Reply:
If upload is moved into `finish` of the writer, we will have no way to simply build an SST in-memory. My original intention is to correspond one implmentation of `SstableWriter` trait with one or more implementations of `SstableBuilderSealer`.

Reply:
Why there is no way to simply build an in-memory SST 🤔? 

The `WriterOutput` of `InMemorySstableWriter` is `Bytes`, and the in-memory SST can be brought out of `CapacitySplitSstableBuilder` via `SealedSstableBuilder`.

If we move upload into `finish`, we can use `JoinHandle` as the `WriterOutput` of `SstableWriter`, and then we join the `JoinHandle` obtained from the `SealedSstableBuilder` after the `CapacitySplitSstableBuilder` is finished.

Reply:
Then we would need a dummy `SstableStore`, since `upload` has to happen.

Reply:
We can store a `SstableStoreRef` in the corresponding `SstableWriter` implementation, and spawn a new task for upload in `finish`, and return the `JoinHandle`.

Reply:
Sure, but in cases such as unit test or benchmarks, we don't need the `SstableStoreRef` in the first place, because we don't need to upload the SSTs. But with `sstable_store.upload` being a part of `finish`, we have to use some dummy `SstableStore`.

Reply:
How about we introduce another writer called `BatchUploadSstableWriter`, instead of using `InMemSstableWriter` for tests, benchmarks and batch upload?

Reply:
> But with sstable_store.upload being a part of finish, we have to use some dummy SstableStore.

`sstable_store.upload` is just part of an implementation of the `finish` of `SstableWriter`. Neither `SstableBuilder` nor `CapacitySplitSstableBuilder` is aware of the `sstable_store.upload`.